src/Controller/BlockAdminController.php

```diff
@@ -86,8 +86,7 @@ public function createAction(Request $request): Response
     {
         $this->admin->checkAccess('create');
 
-        $sharedBlockAdminClass = $this->getParameter('sonata.page.admin.shared_block.class');
-        if (!$this->admin->isChild() && \get_class($this->admin) !== $sharedBlockAdminClass) {
+        if (!$this->admin->isChild()) {
```

Comment:
this was a weird hack to avoid creating a block outside child or shared admin, was better solved with the other solution